test/requests/hover_expectations_test.rb

```diff
@@ -109,7 +109,7 @@ def test_hover_apply_target_correction
       $foo = 1
     RUBY
 
-    lines_with_target_correction = [1, 2, 3, 4]
+    lines_with_target_correction = [0, 1, 2, 3]
```

Comment:
We assert that nothing shows up on those lines due to target correction:
* Lines 1, 2, and 3 were well covered.
* Line 0 was not, so I added coverage for it.
* Line 4 was the newline at the end of the snippet.

So if you target this line, in fact, it's a `ProgramNode`, which is not allowed within `Listeners::Hover::ALLOWED_TARGETS`, so the target is still nil, and in the request, [we return early when it's nil](https://github.com/Shopify/ruby-lsp/blob/e0299e55164d908ff3c09e948cdae85bd36ea30d/lib/ruby_lsp/requests/hover.rb#L59-L60), this is why this test was passing before. 

I changed this line because I noticed it while writing a similar test setup on definition. I was hesitant to commit it in the same batch tbh 😄